### ISSUE TITLE
clang toolset emitting wrong settings in Visual Studio backend. 

### ISSUE THREAD

**Issue by tvandijck** [premake/premake-core#699]
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Clang.targets(210,5): error : Element has an invalid value of "ProgramDatabase".
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Clang.targets(210,5): error : Element has an invalid value of "false".
1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Clang.targets(210,5): error : Element has an invalid value of "Level4".

**Comment by samsinsane** [comment]
Setting `toolset "clang"` doesn't seem to be supported in the VS code. How are you achieving these errors? Could this be a 3rd party module?

**Comment by tvandijck** [comment]
No not a 3prd party module... it just works with Visual Studio 2015 and later there is the c2clang compiler, which results in just another toolset in visual studio. 
![image](https://cloud.githubusercontent.com/assets/451515/25487740/fd5d6350-2b19-11e7-8662-26408fd887d8.png)

**Comment by samsinsane** [comment]
So you're just setting `toolset "v141_clang_c2"`?

**Comment by tvandijck** [comment]
Yup, or in VS2015, v140_clang_c2 I think..

**Comment by samsinsane** [comment]
Right, that makes sense. I was just trying `toolset "clang"`, and just could figure out how you managed to produce these errors haha. If this was to be added into core properly, would you expect to have to specify the `v140`/`v141`? I think `toolset "clang"` would be nice, but since MS calls this "C2/Clang" or "Clang/C2", perhaps we should have the toolset be something similar?

**Comment by tvandijck** [comment]
nah, we can just stick with the "v141_clang_c2" toolset name.. it's explicit and works for VS2017... I have no issues with that actually... obviously, the visual studio action COULD potentially just convert 'clang' into that by default... but that doesn't really bother me much...

the problems are however, that currently if you do, you get options emitted that break the actual build. And interestingly, I'm just looking at Xbone and PS4 support today, and it looks like similar issues will come in play there..

So it turns out that the XML output of the "ClCompile" section in the Visual Studio action is very dependent on the toolset, platform and architecture used.

**Comment by samsinsane** [comment]
I'll have to try using that Clang with MS codegen stuff again. I didn't have much success last time, we had some inline asm from something, which it didn't support. If I get a chance, I'll try to contribute some changes to get it working out of the box. I'll make sure I test it against VS2017 too.

**Comment by TurkeyMan** [comment]
I have an (out of date) module somewhere that supports this, and also the LLVM clang toolset.

**Comment by nickclark2016** [comment]
This is likely very out of date at this point, due to improvements on the MSFT side of the clang toolset in VS2019 and VS2022. Closing for now, unless there is need for support back on VS2017 or older for Clang.
